### ISSUE TITLE
fits2bitmap errors

### ISSUE THREAD

**Issue by cdeil** [astropy/astropy#4453]
I wanted to use the [fits2bitmap](http://astropy.readthedocs.org/en/latest/visualization/index.html?highlight=fits2bitmap#scripts) command line tool you wrote and ran into the following errors.

```
$ fits2bitmap emin_0010_emax_1000.fits.gz 
Traceback (most recent call last):
  File "/Users/deil/Library/Python/3.4/bin/fits2bitmap", line 9, in <module>
    load_entry_point('astropy==1.1.dev14232', 'console_scripts', 'fits2bitmap')()
  File "/Users/deil/Library/Python/3.4/lib/python/site-packages/astropy-1.1.dev14232-py3.4-macosx-10.11-x86_64.egg/astropy/visualization/scripts/fits2bitmap.py", line 138, in main
    power=args.power, asinh_a=args.asinh_a, cmap=args.cmap)
  File "/Users/deil/Library/Python/3.4/lib/python/site-packages/astropy-1.1.dev14232-py3.4-macosx-10.11-x86_64.egg/astropy/visualization/scripts/fits2bitmap.py", line 87, in fits2bitmap
    max_percent=max_percent, percent=percent)
  File "/Users/deil/Library/Python/3.4/lib/python/site-packages/astropy-1.1.dev14232-py3.4-macosx-10.11-x86_64.egg/astropy/visualization/ui.py", line 102, in scale_image
    return (stretch + interval)(image, clip=clip)
  File "/Users/deil/Library/Python/3.4/lib/python/site-packages/astropy-1.1.dev14232-py3.4-macosx-10.11-x86_64.egg/astropy/visualization/transform.py", line 37, in __call__
    return self.transform_2(self.transform_1(values, clip=clip), clip=clip)
  File "/Users/deil/Library/Python/3.4/lib/python/site-packages/astropy-1.1.dev14232-py3.4-macosx-10.11-x86_64.egg/astropy/visualization/interval.py", line 36, in __call__
    values = np.subtract(values, float(vmin))
TypeError: float() argument must be a string or a number, not 'NoneType'
```

It's not clear from the help page what the default `vmin` should be:
```
$ fits2bitmap -h
usage: fits2bitmap [-h] [-e hdu] [-o filename] [--scale SCALE] [--power POWER]
                   [--asinh_a ASINH_A] [--min_cut MIN_CUT] [--max_cut MAX_CUT]
                   [--min_percent MIN_PERCENT] [--max_percent MAX_PERCENT]
                   [--percent PERCENT] [--cmap colormap_name]
                   filename [filename ...]

Create a bitmap file from a FITS image.

positional arguments:
  filename              Path to one or more FITS files to convert

optional arguments:
  -h, --help            show this help message and exit
  -e hdu, --ext hdu     specify the HDU extension number or name
  -o filename           Filename for the output image (Default is a PNG file
                        with the same name as the FITS file)
  --scale SCALE         Type of image scaling ("linear", "sqrt", "power",
                        "log", or "asinh")
  --power POWER         Power index for "power" scaling
  --asinh_a ASINH_A     The value in normalized image where the asinh curve
                        transitions from linear to logarithmic behavior (used
                        only for "asinh" scaling)
  --min_cut MIN_CUT     The pixel value of the minimum cut level
  --max_cut MAX_CUT     The pixel value of the maximum cut level
  --min_percent MIN_PERCENT
                        The percentile value used to determine the minimum cut
                        level
  --max_percent MAX_PERCENT
                        The percentile value used to determine the maximum cut
                        level
  --percent PERCENT     The percentage of the image values used to determine
                        the pixel values of the minimum and maximum cut levels
  --cmap colormap_name  matplotlib color map name
```

But given that it's an option, there should be default and not `None`, no?

The docs also suggest that `--ext` can take a name.
But this doesn't work:
```
$ fits2bitmap emin_0010_emax_1000.fits.gz --ext SIGNIFICANCE
usage: fits2bitmap [-h] [-e hdu] [-o filename] [--scale SCALE] [--power POWER]
                   [--asinh_a ASINH_A] [--min_cut MIN_CUT] [--max_cut MAX_CUT]
                   [--min_percent MIN_PERCENT] [--max_percent MAX_PERCENT]
                   [--percent PERCENT] [--cmap colormap_name]
                   filename [filename ...]
fits2bitmap: error: argument -e/--ext: invalid int value: 'SIGNIFICANCE'
```

Overall more tests / a few examples for `fits2bitmap` would be good.

@larrybradley - Do you have time for this in the coming days / weeks? If no, I could take a stab at fixing these issues and improving tests / docs a bit.

**Comment by cdeil** [comment]
I forgot to say ... the image I'm using is here:
```
wget https://raw.githubusercontent.com/cdeil/fermi-ts-maps/master/emin_0030_emax_0100.fits.gz
```
But I don't think it matters for these issues ... you can use any image.

**Comment by embray** [comment]
I set "priority low" since these are mostly UI bugs, but feel free to change if anyone disagrees.  Definitely should be fixed, and should be easy enough to fix too.

**Comment by larrybradley** [comment]
Thanks, @cdeil.  I'll look at this on Monday.

**Comment by larrybradley** [comment]
@cdeil, I've fixed these issue in #4468.

You were getting the first error because the `fits2bitmap` default extension is `0` and your FITS file (`emin_0030_emax_0100.fits.gz`) does not have any data in the primary extension.  The default `min_cut` is set to the image minimum, which then failed.  The PR (#4468) fixes this by using `fits.getdata()`, which will find the first extension with data if `-e 0`.

PR #4468 also fixes the issue to allow the extension to be either the name or number.  It also fixes an issue with the default output filename for compressed FITS files.

**Comment by cdeil** [comment]
@larrybradley  - Thank you!
